docs/source/llms/tracing/index.rst

```diff
@@ -0,0 +1,558 @@
+
+.. code-block:: python
+
+    from mlflow import MlflowClient
+
+    client = MlflowClient()
+
+    # Start a new trace
+    root_span = client.start_trace("my_trace")
+
+    # The request_id is used for creating additional spans that have a hierarchical association to this root span
+    request_id = root_span.request_id
+
+Adding a Child Span
+^^^^^^^^^^^^^^^^^^^
+
+Once a trace is started, you can add child spans to it. Child spans allow you to break down the trace into smaller, more manageable segments, 
+each representing a specific operation or step within the overall process.
+
+.. code-block:: python
+
+    # Create a child span
+    child_span = client.start_span(
+        name="child_span",
+        request_id=request_id,
+        parent_id=root_span.span_id,
+        inputs={"input_key": "input_value"},
+        attributes={"attribute_key": "attribute_value"},
+    )
+
+Ending a Span
+^^^^^^^^^^^^^
+
+After performing the operations associated with a span, you must end the span explicitly using the ``end_span`` method. Make note of the two required fields 
+that are in the API signature:
+
+- **request_id**: The identifier associated with the root span
+- **span_id**: The identifier associated with the span that is being ended
+
+In order to effectively end a particular span, both the root span (returned from calling ``start_trace``) and the targeted span (returned from calling ``start_span``)
+need to be identified when calling the ``end_span`` API. It's a good practice to ensure that the root span's ``request_id`` is readily accessible from all locations
+that you will be starting or ending spans to ensure that any spans created within the context of a trace are properly associated with the root span. 
+
+.. note::
+    Spans created via the Client API will need to be terminated manually. Ensure that all spans that have been started with the ``start_span`` API 
+    have been ended with the ``end_span`` API.
+
+.. code-block:: python
+
+    # End the child span
+    client.end_span(
+        request_id=request_id,
+        span_id=child_span.span_id,
+        outputs={"output_key": "output_value"},
+        attributes={"custom_attribute": "value"},
+    )
+
+Ending a Trace
+^^^^^^^^^^^^^^
+
+To complete the trace, end the root span using the ``end_trace`` method. This will also ensure that all associated child spans are properly closed.
+
+.. code-block:: python
+
+    # End the root span (trace)
+    client.end_trace(
+        request_id=request_id,
+        outputs={"final_output_key": "final_output_value"},
+        attributes={"token_usage": "1174"},
+    )
+
+Searching and Retrieving Traces
+-------------------------------
+
+Searching for Traces
+^^^^^^^^^^^^^^^^^^^^
+
+You can search for traces based on various criteria using the ``search_traces`` method. This method allows you to filter traces by experiment IDs, 
```

Comment:
```suggestion
You can search for traces based on various criteria using the :py:meth:`mlflow.client.MlflowClient.search_traces` method. This method allows you to filter traces by experiment IDs, 
```

Can we add link to other methods like delete_traces too? The docstring has rich description and examples.

Reply:
All have been updated